### PR TITLE
(PA-159) libxml2 and libxslt support for huaweios

### PR DESCRIPTION
**Pull request by ScottGarman**

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
:+1:

File: configs/components/libxslt.rb [hunk 1]
```diff
@@ -8,6 +8,13 @@
   if platform.is_aix?
     pkg.build_requires "http://pl-build-tools.delivery.puppetlabs.net/aix/#{platform.os_version}/ppc/pl-gcc-5.2.0-1.aix#{platform.os_version}.ppc.rpm"
     pkg.environment "PATH" => "/opt/pl-build-tools/bin:$$PATH"
+  elsif platform.is_huaweios?
+    pkg.environment "PATH" => "/opt/pl-build-tools/bin:$$PATH:#{settings[:bindir]}"
+    pkg.environment "CFLAGS" => settings[:cflags]
+    pkg.environment "LDFLAGS" => settings[:ldflags]
+
+    # libxslt is picky about manually specifying the build host
+    build = "--build x86_64-unknown-linux-gnu"
```
Comment (melissa) [inline configs/components/libxslt.rb]: Do we need this for any other cross compiles that are happening?

Comment (ScottGarman) [inline configs/components/libxslt.rb]: @melissa I don't know why the solaris sparc cross-build doesn't need this, but I'm not inclined to change it on other platforms for the time being.

Comment (melissa) [inline configs/components/libxslt.rb]: totes fair. I was just wondering. If it works, then we're probably good

**Pull request merged**